File: Flix.Backend/Flix.Services/Database/MovieRecommendation.cs
using System.ComponentModel.DataAnnotations;

namespace Flix.Services.Database
{
    public class MovieRecommendation
    {
        [Key]
        public int Id {get; set;}
        public int MovieId {get; set;}
        public Movie Movie { get; set; } = null!;
        public int RecommendedMovieId {get; set;}
        public Movie RecommendedMovie {get; set;} = null!;
        [Required]
        public float Score { get; set; }
    }
}

File: Flix.UI/flix_mobile/lib/models/movie_recommendation.dart
import 'package:flix_mobile/enums/recommendation_source.dart';
import 'package:flix_mobile/models/movie.dart';
import 'package:json_annotation/json_annotation.dart';

part 'movie_recommendation.g.dart';

@JsonSerializable()
class MovieRecommendation {
  MovieRecommendation(
    this.id,
    this.movieId,
    this.movie,
    this.recommendedMovieId,
    this.recommendedMovie,
    this.score,
    this.source,
    this.reason,
  );

  final int? id;

  final int? movieId;
  final Movie? movie;

  final int? recommendedMovieId;
  final Movie? recommendedMovie;

  final double? score;

  final RecommendationSource? source;
  final String? reason;

  factory MovieRecommendation.fromJson(Map<String, dynamic> json) =>
      _$MovieRecommendationFromJson(json);

  Map<String, dynamic> toJson() => _$MovieRecommendationToJson(this);
}
